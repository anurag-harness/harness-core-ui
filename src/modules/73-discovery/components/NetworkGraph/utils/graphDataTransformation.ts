/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { Node, Edge } from 'reactflow'
import { NodeTypes, nodeGroupOptions, nodeOptions } from '@discovery/components/NetworkGraph/constants'
import type {
  ApiCreateNetworkMapRequest,
  ApiListDiscoveredServiceConnection,
  ApiListDiscoveredService
} from 'services/servicediscovery'
import type {
  NetworkMapEdgeData,
  NetworkMapNodeData,
  ServiceEdgeData,
  ServiceNodeData
} from '@discovery/components/NetworkGraph/types'

export function getGraphNodesFromServiceList(
  serviceList: ApiListDiscoveredService | null
): Node<ServiceNodeData, NodeTypes>[] {
  if (!serviceList) return []

  const namespaces = new Set<string>()
  const graphNodes: Node<ServiceNodeData, NodeTypes>[] = []
  serviceList.items?.map(service => {
    const namespace = service.spec.kubernetes?.namespace
    if (namespace && !namespaces.has(namespace)) {
      namespaces.add(namespace)
    }

    if (service.id)
      graphNodes.push({
        ...nodeOptions,
        id: service.id,
        data: { ...service, name: service.spec.kubernetes?.name ?? 'service' },
        parentNode: namespace
      })
  })

  namespaces.forEach(value => {
    graphNodes.push({
      ...nodeGroupOptions,
      id: value,
      data: { id: value, name: value }
    })
  })

  return graphNodes
}

export function getGraphEdgesFromServiceConnections(
  connectionList: ApiListDiscoveredServiceConnection | null
): Edge<ServiceEdgeData>[] {
  if (!connectionList) return []

  const graphEdges: Edge<ServiceEdgeData>[] = []

  connectionList.items?.map(connection => {
    if (connection.id && connection.sourceID && connection.destinationID)
      graphEdges.push({
        id: connection.id,
        source: connection.sourceID,
        target: connection.destinationID,
        data: { parentNode: connection.sourceNamespace, ...connection }
      })
  })

  return graphEdges
}

export function getGraphNodesFromNetworkMap(
  networkMap: ApiCreateNetworkMapRequest | undefined
): Node<NetworkMapNodeData, NodeTypes>[] {
  if (!networkMap) return []

  const namespaces = new Set<string>()
  const graphNodes: Node<NetworkMapNodeData, NodeTypes>[] = []

  networkMap.resources?.map(service => {
    const namespace = service.kubernetes?.namespace
    if (namespace && !namespaces.has(namespace)) {
      namespaces.add(namespace)
    }

    if (service.id)
      graphNodes.push({
        ...nodeOptions,
        type: NodeTypes.NetworkMapHexagon,
        id: service.id,
        data: service,
        parentNode: namespace
      })
  })

  namespaces.forEach(value => {
    graphNodes.push({
      ...nodeGroupOptions,
      id: value,
      data: { id: value, name: value, kind: 'discoveredservice' }
    })
  })

  return graphNodes
}

export function getGraphEdgesFromNetworkMap(
  networkMap: ApiCreateNetworkMapRequest | undefined
): Edge<NetworkMapEdgeData>[] {
  if (!networkMap) return []

  const graphEdges: Edge<NetworkMapEdgeData>[] = []

  networkMap.connections?.map(connection => {
    if (connection.from?.id && connection.to?.id)
      graphEdges.push({
        id: `${connection.from.id}-${connection.to.id}`,
        source: connection.from.id,
        target: connection.to.id,
        data: { ...connection, parentNode: connection.from.kubernetes?.namespace }
      })
  })

  return graphEdges
}
