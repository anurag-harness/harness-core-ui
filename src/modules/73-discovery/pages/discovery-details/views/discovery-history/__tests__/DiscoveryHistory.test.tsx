/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import * as servicediscovery from 'services/servicediscovery'
import { accountPathProps, modulePathProps, projectPathProps } from '@common/utils/routeUtils'
import DiscoveryHistory from '../DiscoveryHistory'

jest.useFakeTimers({ advanceTimers: true })

const paginationProps = {
  all: true,
  index: 0,
  totalItems: 1,
  totalPages: 1
}

const mockListInstallation: servicediscovery.DatabaseInstallationCollection = {
  id: '123',
  agentID: '1234',
  createdAt: '2023-06-21T13:17:41.491Z',
  createdBy: 'VgWXxi_6TdqAyplTQMg4CQ',
  updatedAt: '2023-06-21T13:17:41.491Z',
  delegateID: '1234',
  updatedBy: ''
}

const mockInstallationList: servicediscovery.ApiListInstallationResponse = {
  items: [mockListInstallation],
  page: paginationProps
}

const fetchListInstallation = jest.fn(() => {
  return Object.create(mockInstallationList)
})

jest.mock('services/servicediscovery', () => ({
  useListInstallation: jest.fn().mockImplementation(() => {
    return { data: mockInstallationList, refetch: fetchListInstallation, error: null, loading: false }
  })
}))

jest.mock('services/logs', () => ({
  useGetToken: jest.fn(() => ({ data: 'logs_token' })),
  logBlobPromise: jest.fn(() => Promise.resolve({}))
}))

const PATH = routes.toDiscovery({ ...accountPathProps, ...projectPathProps, ...modulePathProps })
const PATH_PARAMS = {
  accountId: 'accountId',
  orgIdentifier: 'default',
  projectIdentifier: 'Discovery_Test',
  module: 'chaos',
  dAgentId: '1234'
}

describe('<DiscoveryHistory /> tests', () => {
  beforeEach(() => jest.clearAllMocks())

  test('should match snapshot', async () => {
    const { container } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryHistory />
      </TestWrapper>
    )
    expect(servicediscovery.useListInstallation).toBeCalled()
    expect(container).toMatchSnapshot()
  })

  test('should render loading view correctly', async () => {
    jest.spyOn(servicediscovery, 'useListInstallation').mockImplementation((): any => {
      return {
        data: undefined,
        loading: true
      }
    })

    const { container } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryHistory />
      </TestWrapper>
    )
    expect(servicediscovery.useListInstallation).toBeCalled()

    expect(container).toMatchSnapshot()
  })

  test('should render error view correctly', async () => {
    jest.spyOn(servicediscovery, 'useListInstallation').mockImplementation((): any => {
      return {
        data: undefined,
        loading: false,
        error: {
          message: 'some error'
        }
      }
    })

    const { container, getByText } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryHistory />
      </TestWrapper>
    )
    expect(servicediscovery.useListInstallation).toBeCalled()

    expect(container).toMatchSnapshot()

    expect(getByText('discovery.discoveryDetails.tabTitles.history')).toBeInTheDocument()
  })
})
