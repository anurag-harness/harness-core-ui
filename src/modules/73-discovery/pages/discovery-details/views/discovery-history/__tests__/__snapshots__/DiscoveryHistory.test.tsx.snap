// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscoveryHistory /> tests should match snapshot 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main"
    style="width: 100%; height: 100%;"
  >
    <div
      class="StyledProps--font StyledProps--main"
      style="width: 35%;"
    >
      <div
        class="Layout--vertical StyledProps--main"
        style="height: 100%;"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between"
          style="padding: 10px 30px;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-card-title"
          >
            discovery.discoveryDetails.tabTitles.history
          </p>
          <div>
            <button
              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-repeat StyledProps--main"
                icon="repeat"
              >
                <svg
                  data-icon="repeat"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    repeat
                  </desc>
                  <path
                    d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
            <button
              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="main-sort"
              >
                <svg
                  height="16"
                  viewBox="0 0 18 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M17.125 11.415l-3.713 3.427a.618.618 0 01-.825 0l-3.712-3.427a.51.51 0 010-.761.618.618 0 01.825 0l2.717 2.508V1.538c0-.297.26-.538.583-.538.322 0 .583.241.583.538v11.624l2.717-2.508a.618.618 0 01.825 0 .51.51 0 010 .761zM.875 4.584l3.713-3.426a.618.618 0 01.824 0l3.713 3.426a.51.51 0 010 .762.618.618 0 01-.825 0L5.583 2.838v11.623c0 .298-.26.539-.583.539-.322 0-.583-.241-.583-.539V2.838L1.7 5.346a.618.618 0 01-.825 0 .51.51 0 010-.762z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="selectableCard"
        >
          <div
            class="bp3-card bp3-interactive bp3-elevation-0 Card--card Card--selected Card--interactive"
            data-index="0"
            tabindex="0"
          >
            <div
              class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-body2"
                style="font-weight: 500;"
              >
                Jun 21, 01:17 PM
              </p>
              <div
                class="StyledProps--font StyledProps--main"
                style="padding: 2px 6px; border-radius: 4px;"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--color StyledProps--color-grey700"
                  style="font-weight: 600;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main pagination Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
        >
          <span
            class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
          >
            1
             of 
            1
          </span>
          <div
            class="Layout--horizontal StyledProps--main"
          >
            <button
              aria-label="Prev"
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonLeft Button--with-current-color Button--withLeftIcon Button--size Button--small"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-arrow-left StyledProps--main StyledProps--padding-right-xsmall"
                icon="arrow-left"
              >
                <svg
                  data-icon="arrow-left"
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <desc>
                    arrow-left
                  </desc>
                  <path
                    d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                Prev
              </span>
            </button>
            <button
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--selected Button--with-current-color"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                1
              </span>
            </button>
            <button
              aria-label="Next"
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonRight Button--with-current-color Button--withRightIcon Button--size Button--small"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Next
              </span>
              <span
                class="bp3-icon bp3-icon-arrow-right StyledProps--main"
                icon="arrow-right"
              >
                <svg
                  data-icon="arrow-right"
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <desc>
                    arrow-right
                  </desc>
                  <path
                    d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline"
          >
            <p
              class="StyledProps--font StyledProps--main"
            >
              Show
            </p>
            <div
              class="bp3-popover-wrapper DropDown--main Pagination--pageSizeDropdown"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class=""
                >
                  <div
                    class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                    data-testid="dropdown-button"
                    style="width: 60px;"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                    >
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                        data-testid="dropdown-value"
                        style="--text-line-clamp: 1;"
                      >
                        20
                      </p>
                    </div>
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                    >
                      <span
                        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                        data-icon="main-chevron-down"
                      >
                        <svg
                          height="8"
                          viewBox="0 0 14 8"
                          width="8"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                            fill="#4d5969"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <p
              class="StyledProps--font StyledProps--main"
            >
              per page
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
      style="width: 65%; height: 100%;"
    >
      <div
        class="Layout--vertical StyledProps--main"
        style="height: 100%;"
      >
        <div
          class="StyledProps--font StyledProps--main logsTab StyledProps--background-black"
          style="width: 100%; height: 100%;"
        >
          <div
            class="bp3-tabs Tabs--main Tabs--horizontal"
          >
            <div
              class="bp3-tab-list"
              role="tablist"
            >
              <div
                class="bp3-tab-indicator-wrapper"
                style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
              >
                <div
                  class="bp3-tab-indicator"
                />
              </div>
              <div
                aria-controls="bp3-tab-panel_DiscoveredHistoryTab_cluster"
                aria-disabled="false"
                aria-expanded="true"
                aria-selected="true"
                class="bp3-tab"
                data-tab-id="cluster"
                id="bp3-tab-title_DiscoveredHistoryTab_cluster"
                role="tab"
                tabindex="0"
              >
                common.cluster
              </div>
              <div
                aria-controls="bp3-tab-panel_DiscoveredHistoryTab_node"
                aria-disabled="false"
                aria-expanded="false"
                aria-selected="false"
                class="bp3-tab"
                data-tab-id="node"
                id="bp3-tab-title_DiscoveredHistoryTab_node"
                role="tab"
                tabindex="0"
              >
                common.node
              </div>
            </div>
            <div
              aria-hidden="false"
              aria-labelledby="bp3-tab-title_DiscoveredHistoryTab_cluster"
              class="bp3-tab-panel"
              id="bp3-tab-panel_DiscoveredHistoryTab_cluster"
              role="tabpanel"
            >
              <div
                class="logViewer logContainer"
                style="--char-size: 2ch;"
              >
                <div
                  class="bp3-spinner"
                >
                  <div
                    class="bp3-spinner-animation"
                  >
                    <svg
                      height="50"
                      stroke-width="8.00"
                      viewBox="1.00 1.00 98.00 98.00"
                      width="50"
                    >
                      <path
                        class="bp3-spinner-track"
                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                      />
                      <path
                        class="bp3-spinner-head"
                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                        pathLength="280"
                        stroke-dasharray="280 280"
                        stroke-dashoffset="210"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DiscoveryHistory /> tests should render error view correctly 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main"
    style="width: 100%; height: 100%;"
  >
    <div
      class="StyledProps--font StyledProps--main"
      style="width: 35%;"
    >
      <div
        class="Layout--vertical StyledProps--main"
        style="height: 100%;"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between"
          style="padding: 10px 30px;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-card-title"
          >
            discovery.discoveryDetails.tabTitles.history
          </p>
          <div>
            <button
              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-repeat StyledProps--main"
                icon="repeat"
              >
                <svg
                  data-icon="repeat"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    repeat
                  </desc>
                  <path
                    d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
            <button
              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="main-sort"
              >
                <svg
                  height="16"
                  viewBox="0 0 18 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M17.125 11.415l-3.713 3.427a.618.618 0 01-.825 0l-3.712-3.427a.51.51 0 010-.761.618.618 0 01.825 0l2.717 2.508V1.538c0-.297.26-.538.583-.538.322 0 .583.241.583.538v11.624l2.717-2.508a.618.618 0 01.825 0 .51.51 0 010 .761zM.875 4.584l3.713-3.426a.618.618 0 01.824 0l3.713 3.426a.51.51 0 010 .762.618.618 0 01-.825 0L5.583 2.838v11.623c0 .298-.26.539-.583.539-.322 0-.583-.241-.583-.539V2.838L1.7 5.346a.618.618 0 01-.825 0 .51.51 0 010-.762z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="selectableCard"
        />
        <div
          class="Layout--horizontal StyledProps--main pagination Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
        >
          <span
            class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
          >
            0
             of 
            0
          </span>
          <div
            class="Layout--horizontal StyledProps--main"
          >
            <button
              aria-label="Prev"
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonLeft Button--with-current-color Button--withLeftIcon Button--size Button--small"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-arrow-left StyledProps--main StyledProps--padding-right-xsmall"
                icon="arrow-left"
              >
                <svg
                  data-icon="arrow-left"
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <desc>
                    arrow-left
                  </desc>
                  <path
                    d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                Prev
              </span>
            </button>
            <button
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--selected Button--with-current-color"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                1
              </span>
            </button>
            <button
              aria-label="Next"
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonRight Button--with-current-color Button--withRightIcon Button--size Button--small"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Next
              </span>
              <span
                class="bp3-icon bp3-icon-arrow-right StyledProps--main"
                icon="arrow-right"
              >
                <svg
                  data-icon="arrow-right"
                  height="12"
                  viewBox="0 0 16 16"
                  width="12"
                >
                  <desc>
                    arrow-right
                  </desc>
                  <path
                    d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline"
          >
            <p
              class="StyledProps--font StyledProps--main"
            >
              Show
            </p>
            <div
              class="bp3-popover-wrapper DropDown--main Pagination--pageSizeDropdown"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class=""
                >
                  <div
                    class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                    data-testid="dropdown-button"
                    style="width: 60px;"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                    >
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                        data-testid="dropdown-value"
                        style="--text-line-clamp: 1;"
                      >
                        20
                      </p>
                    </div>
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                    >
                      <span
                        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                        data-icon="main-chevron-down"
                      >
                        <svg
                          height="8"
                          viewBox="0 0 14 8"
                          width="8"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                            fill="#4d5969"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <p
              class="StyledProps--font StyledProps--main"
            >
              per page
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
      style="width: 65%; height: 100%;"
    >
      <div
        class="Layout--vertical StyledProps--main"
        style="height: 100%;"
      >
        <div
          class="StyledProps--font StyledProps--main logsTab StyledProps--background-black"
          style="width: 100%; height: 100%;"
        >
          <div
            class="bp3-tabs Tabs--main Tabs--horizontal"
          >
            <div
              class="bp3-tab-list"
              role="tablist"
            >
              <div
                class="bp3-tab-indicator-wrapper"
                style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
              >
                <div
                  class="bp3-tab-indicator"
                />
              </div>
              <div
                aria-controls="bp3-tab-panel_DiscoveredHistoryTab_cluster"
                aria-disabled="false"
                aria-expanded="true"
                aria-selected="true"
                class="bp3-tab"
                data-tab-id="cluster"
                id="bp3-tab-title_DiscoveredHistoryTab_cluster"
                role="tab"
                tabindex="0"
              >
                common.cluster
              </div>
              <div
                aria-controls="bp3-tab-panel_DiscoveredHistoryTab_node"
                aria-disabled="false"
                aria-expanded="false"
                aria-selected="false"
                class="bp3-tab"
                data-tab-id="node"
                id="bp3-tab-title_DiscoveredHistoryTab_node"
                role="tab"
                tabindex="0"
              >
                common.node
              </div>
            </div>
            <div
              aria-hidden="false"
              aria-labelledby="bp3-tab-title_DiscoveredHistoryTab_cluster"
              class="bp3-tab-panel"
              id="bp3-tab-panel_DiscoveredHistoryTab_cluster"
              role="tabpanel"
            >
              <div
                class="logViewer logContainer"
                style="--char-size: 2ch;"
              >
                <div
                  class="bp3-spinner"
                >
                  <div
                    class="bp3-spinner-animation"
                  >
                    <svg
                      height="50"
                      stroke-width="8.00"
                      viewBox="1.00 1.00 98.00 98.00"
                      width="50"
                    >
                      <path
                        class="bp3-spinner-track"
                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                      />
                      <path
                        class="bp3-spinner-head"
                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                        pathLength="280"
                        stroke-dasharray="280 280"
                        stroke-dashoffset="210"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DiscoveryHistory /> tests should render loading view correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody"
    data-name="page-body"
    data-testid="page-body"
  >
    <div
      class="StyledProps--font StyledProps--main PageSpinner--spinner StyledProps--flex StyledProps--flex-align-center-center"
      data-testid="page-spinner"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main PageSpinner--content"
        style="align-items: center; width: 500px;"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
          data-icon="steps-spinner"
        >
          <svg
            height="32"
            viewBox="0 0 128 128"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                fill="#ff8e02"
              />
              <path
                d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                fill="#ffddb3"
                fill-opacity="0.3"
              />
              <path
                d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                fill="#ffd29a"
                fill-opacity="0.4"
              />
              <path
                d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                fill="#ffc680"
                fill-opacity="0.5"
              />
              <path
                d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                fill="#ffbb67"
                fill-opacity="0.6"
              />
              <path
                d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                fill="#ffb04d"
                fill-opacity="0.7"
              />
              <path
                d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                fill="#ffa535"
                fill-opacity="0.8"
              />
              <path
                d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                fill="#ff991b"
                fill-opacity="0.9"
              />
              <animatetransform
                attributeName="transform"
                calcMode="discrete"
                dur="1200ms"
                repeatCount="indefinite"
                type="rotate"
                values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
              />
            </g>
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
        >
          Loading, please wait...
        </p>
      </div>
    </div>
  </div>
</div>
`;
